src/main.rs

```diff
@@ -2,56 +2,69 @@
 // TODO: support publishing to S-Miles cloud, too
 
 mod inverter;
-mod mqtt;
-mod protos;
 mod logging;
+mod metric_collector;
+mod mqtt;
+mod mqtt_config;
 mod mqtt_schemas;
+mod protos;
+mod simple_mqtt;
 
 use crate::inverter::Inverter;
-use crate::mqtt::{MetricCollector, Mqtt};
 use crate::logging::init_logger;
+use crate::metric_collector::MetricCollector;
+use crate::mqtt::Mqtt;
+use crate::simple_mqtt::SimpleMqtt;
 
+use mqtt_config::MqttConfig;
+use serde_derive::Deserialize;
+use std::fs;
 use std::thread;
 use std::time::Duration;
 
-use clap::Parser;
-use log::info;
+use log::{error, info};
 use protos::hoymiles::RealData;
 
-#[derive(Parser)]
-struct Cli {
+#[derive(Debug, Deserialize)]
+struct Config {
     inverter_host: String,
-    mqtt_broker_host: String,
-    mqtt_username: Option<String>,
-    mqtt_password: Option<String>,
-    #[clap(default_value = "1883")]
-    mqtt_broker_port: u16,
+    home_assistent: Option<MqttConfig>,
```

Comment:
typo: `home_assistant` (needs to be adapted in config.yaml and run.sh as well)

Reply:
good catch!